### PR TITLE
CIV-6421 court location name

### PR DESCRIPTION
**Pull request by MMNycz**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-6421

### Change description ###

Add preferred court location on DQ and Response Sealed Claim form Templates 

Update templates name:
 
CV-SPC-HRN-ENG-01046 -> CV-SPC-HRN-ENG-01274
CV-SPC-HRN-ENG-01062 -> CV-SPC-HRN-ENG-01275
CV-SPC-HRN-ENG-01064 -> CV-SPC-HRN-ENG-01276
CV-SPC-HRN-ENG-01065 -> CV-SPC-HRN-ENG-01277
CV-SPC-HRN-ENG-01209 -> CV-SPC-HRN-ENG-01278
CV-SPC-HRN-ENG-01246 -> CV-SPC-HRN-ENG-01279
 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapperTest.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import uk.gov.hmcts.reform.civil.model.SolicitorOrganisationDetails;
 import uk.gov.hmcts.reform.civil.model.referencedata.response.LocationRefData;
 import uk.gov.hmcts.reform.civil.model.robotics.NoticeOfChange;
+import uk.gov.hmcts.reform.civil.model.referencedata.response.LocationRefData;
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapperTest.java:24]: this feels like duplicate import here

Comment (MMNycz) [reply src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapperTest.java]: You are better than checkstyle ;) I fix it.

**Review by asthamalviya — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java [hunk 2]
```diff
@@ -497,15 +503,23 @@ private FurtherInformation getFurtherInformation(DQ dq, CaseData caseData) {
 
     private RequestedCourt getRequestedCourt(DQ dq) {
         RequestedCourt rc = dq.getRequestedCourt();
-        if (rc == null) {
+        if (rc != null && null != rc.getCaseLocation()) {
```
Comment (hgulv1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java:506]: can we have consistency in the null checks please so both null != x

File: src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
 import uk.gov.hmcts.reform.civil.model.robotics.RoboticsCaseData;
 import uk.gov.hmcts.reform.civil.model.robotics.Solicitor;
 import uk.gov.hmcts.reform.civil.service.OrganisationService;
+import uk.gov.hmcts.reform.civil.service.referencedata.LocationRefDataService;
```
Comment (hgulv1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java:24]: is this import used anywhere?

File: src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java [hunk 4]
```diff
@@ -51,6 +52,7 @@
 import static uk.gov.hmcts.reform.civil.service.robotics.utils.RoboticsDataUtil.RESPONDENT2_ID;
 import static uk.gov.hmcts.reform.civil.service.robotics.utils.RoboticsDataUtil.RESPONDENT_SOLICITOR_ID;
 import static uk.gov.hmcts.reform.civil.service.robotics.utils.RoboticsDataUtil.RESPONDENT2_SOLICITOR_ID;
+import static uk.gov.hmcts.reform.civil.service.robotics.utils.RoboticsDataUtil.CIVIL_COURT_TYPE_ID;
```
Comment (hgulv1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java:55]: is this import used anywhere?

**Review by hgulv1 — CHANGES_REQUESTED** [review]
Few minor issues.

File: src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java (hunk 3 above)
Comment (MMNycz) [reply src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsDataMapper.java:24]: Resolved - I do not know why it is add double import and checkstyle do not catch it.

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java (hunk 2 above)
Comment (MMNycz) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java]: done

**Review by hgulv1 — APPROVED** [review]

**Review by asthamalviya — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java [hunk 5]
```diff
@@ -495,17 +499,25 @@ private FurtherInformation getFurtherInformation(DQ dq, CaseData caseData) {
...
+                .getCourtLocationsByEpimmsId(
+                    authorisation,
+                    rc.getCaseLocation().getBaseLocation()
+                ));
             return RequestedCourt.builder()
                 .requestHearingAtSpecificCourt(YES)
+                .responseCourtName(courtLocations.isEmpty() ? null :
+                                       courtLocations.stream()
+                                           .filter(id -> id.getCourtTypeId().equals(CIVIL_COURT_TYPE_ID))
+                                           .collect(Collectors.toList()).get(0).getCourtName())
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java:515]: can we check for potential null pointer here as after filter there might not be any record left

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimResponseFormGeneratorForSpec.java [hunk 6]
```diff
@@ -32,24 +35,44 @@
...
+                requestedCourt));
         SealedClaimResponseFormForSpec.SealedClaimResponseFormForSpecBuilder builder
             = SealedClaimResponseFormForSpec.builder()
             .referenceNumber(caseData.getLegacyCaseReference())
             .caseName(DocmosisTemplateDataUtils.toCaseName.apply(caseData))
             .whyDisputeTheClaim(caseData.getDetailsOfWhyDoesYouDisputeTheClaim())
-            .statementOfTruth(caseData.getRespondent1DQ().getRespondent1DQStatementOfTruth());
+            .hearingCourtLocation(courtLocations.isEmpty() ? null : courtLocations.stream()
+                .filter(id -> id.getCourtTypeId().equals(CIVIL_COURT_TYPE_ID))
+                .collect(Collectors.toList()).get(0).getCourtName())
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimResponseFormGeneratorForSpec.java:74]: same here please

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java (hunk 5 above)
Comment (MMNycz) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java]: I make changes and make some manual test.

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimResponseFormGeneratorForSpec.java (hunk 6 above)
Comment (MMNycz) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimResponseFormGeneratorForSpec.java]: I make changes and make some manual test.

**Review by sabahirfan — APPROVED** [review]

**Review by jeejavr — DISMISSED** [review]

**Review by asthamalviya — APPROVED** [review]

**Comment by MMNycz** [comment]
Matt - first you have to merge civil-ccd-definition PR-2703  and in Civil-Service revert ccdBranch pointing to master in Jenkinsfile_CNP.

**Comment by MMNycz** [comment]
It was green when point to new ccdBranch.
image

No I revert it back to master

**Review by asthamalviya — APPROVED** [review]

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**